app/services/fact_importer.rb

```diff
@@ -112,6 +115,6 @@ def normalize(facts)
   end
 
   def db_facts
-    @db_facts ||= host.fact_values.eager_load(:fact_name).where("fact_names.type = '#{fact_name_class}'").index_by(&:name)
+    host.fact_values.eager_load(:fact_name).where("fact_names.type = '#{fact_name_class}'")
```

Comment:
Is there any reason not to use `.joins` (for an inner join) instead of `.eager_load` (which generates a left outer join)?